src/main/java/com/minecolonies/client/gui/WindowTownHall.java

```diff
@@ -221,8 +275,72 @@ public WindowTownHall(BuildingTownHall.View townHall)
         registerButton(BUTTON_CHANGE_SPEC, this::doNothing);
         registerButton(BUTTON_TOGGLE_JOB, this::toggleHiring);
 
+        registerButton("up", this::updatePriority);
+        registerButton("down", this::updatePriority);
+        registerButton("delete", this::deleteWorkOrder);
     }
 
+    /**
+     * On Button click update the priority.
+     * @param button the clicked button.
+     */
+    private void updatePriority(Button button)
+    {
+        Label idLabel = (Label)button.getParent().getChildren().get(HIDDEN_ID_POSITION);
+        int id = Integer.valueOf(idLabel.getLabelText());
+        String buttonLabel = button.getID();
+
+        for (int i = 0; i < workOrders.size(); i++)
+        {
+            WorkOrderView workOrder = workOrders.get(i);
+            if (workOrder.getId() == id)
+            {
+                if(buttonLabel.equals("up"))
+                {
+                    if (i-1 >= 0)
+                    {
+                        workOrder.setPriority(workOrders.get(i-1).getPriority());
+                        workOrders.get(i-1).setPriority(workOrder.getPriority());
```

Comment:
I slightly changed the logic. Now when I click up or down the current workOrder changes it's priority with the next one. When I had the previous way of computing it a down click may've put the workOrder 2 or 3 positions down.

Example:
2
1
1
0

If I demote 2 it may get to the second last position because it has the same priority as the other ones now.

I asked Peter is this is a problem and he preferred not to have it.